src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -1448,6 +1449,32 @@ void goto_instrument_parse_optionst::instrument_goto_program()
       throw 0;
   }
 
+  if(
+    cmdline.isset("replace-function-body") ||
+    cmdline.isset("generate-function-body"))
+  {
+    if(
+      cmdline.isset("replace-function-body") &&
+      cmdline.isset("generate-function-body"))
+    {
+      throw "can only use one of --replace-function-body "
+        "or --generate-function-body";
```

Comment:
Why are these mutually exclusive? To me, it seems ok to request generating bodies for those that never had one in the first place, and also add some more to that list (of bodiless functions) via replace-function-body.

Reply:
I don't like changing how replace-function-body works depending on whether or not there had also been a generate-function-body in the same command line. But if you want something like this maybe I need to rethink the way I currently do the command line argument parsing, could you tell me how you would want to use this from the command line?

Reply:
I thought that "replace-function-body" operates on those functions that previously had a function body, while "generate-function-body" will work on those that don't. Somehow I must be getting this wrong, as these are disjoint sets and thus there wouldn't be any interaction. Would you mind clarifying?

Reply:
@tautschnig `replace-function-body` does "replace" the body of the function with what's specified in `replace-function-body-options`, regardless of whether or not the function has a body. Generate works the same as replace but skips over functions that already have a body.

I did it like that because I thought there might be reasons for why you'd want to only generate bodies for functions that don't have a body yet, but I couldn't think of a reason for why you'd want to replace function  bodies of functions that do already have a body but at the same time leave bodyless functions alone.

I could make these completely separate like you suggested, would that be preferable?

Reply:
I think subtle interaction is dangerous (and hence you disallowed it). How about making the steps very explicit by leaving in place/promoting `remove-function-body`, dropping `replace-function-body`, and leaving all remaining work to `generate-function-body` (including all the options currently specifiable via `replace-function-body-options`)?

Using `remove-function-body` together with `generate-function-body` would then be perfectly permissible, with the sequence 1) remove and 2) generate.

Reply:
@martin-cs @peterschrammel Do you agree that this is a good way to do this? I'm not opposed to doing it this way.

Reply:
\<marketing\>We would thus not have any surprises/disruption for existing users of `remove-function-body`.\</marketing\>